modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java

```diff
@@ -409,6 +413,26 @@ public void testPostCustomObjectEntryWithNestedCustomObjectEntriesInOneToManyRel
 			_OBJECT_FIELD_NAME_2, _NEW_OBJECT_FIELD_VALUE_2);
 	}
 
+	@Test
+	public void testPostCustomObjectEntryWithWrongNestedCustomObjectEntriesInRelationship()
+		throws Exception {
+
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				"com.liferay.portal.vulcan.internal.jaxrs.exception.mapper." +
+					"WebApplicationExceptionMapper",
+				LoggerTestUtil.WARN)) {
+
+			_testPostCustomObjectEntryWithWrongNestedCustomObjectEntriesInManyToManyRelationship(
```

Comment:
To be honest, I don't know if creating three private methods make sense... We are duplicating code constantly.

Please, think about a way to avoid duplicating. Maybe we could create a private method to create a JSONObject / ObjectEntry based on the type of element we want to create